sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReceiveLinkProcessor.java

```diff
@@ -571,21 +562,28 @@ private int getCreditsToAdd(int linkCredits) {
         }
 
         final int creditsToAdd;
-        if (messageQueue.isEmpty() && !hasBackpressure) {
-            creditsToAdd = prefetch;
+        final int expectedTotalCredit;
+        if (prefetch == 0) {
+            if (r <= Integer.MAX_VALUE) {
+                expectedTotalCredit = (int) r;
+            } else {
+                expectedTotalCredit = Integer.MAX_VALUE;
```

Comment:
How we get this in this case where  credit will need to be `Integer.MAX_VALUE` ? And should it be max ?

Reply:
This won't really happen in reality. I still leave it there because Integer.MAX_VALUE is the largest allowed link credits.
For async client, receiveMessages() calls "return receiveMessagesNoBackPressure().limitRate(1, 0);". This will request one by one from the link processor, even though the user's request has no back pressure.

For sync client, the sync subscriber has back pressure. The request count uses the  the argument of method receiveMessages(int maxMessages). It's at most Integer.MAX_VALUE.

Reply:
So does this justify In that case should we log point to help us debug this if needed.

Reply:
Added the code comments.